idaes/core/util/model_statistics.py

```diff
@@ -115,6 +116,104 @@ def activated_blocks_set(block):
     return block_set
 
 
+def greybox_block_set(block):
+    """
+    Function to return ComponentSet of all Greybox Blocks components in a
+    model.
+
+    Args:
+        block : model to be studied
+
+    Returns:
+        A ComponentSet including all GreyBox Block components in block
+        (including block itself)
+    """
+    block_set = ComponentSet()
+    for grey_box in activated_block_component_generator(
+        block, ctype=ExternalGreyBoxBlock
+    ):
+        block_set.add(grey_box)
+
+    return block_set
+
+
+def activated_greybox_block_set(block):
+    """
+    Function to return ComponentSet of activated Greybox Blocks components in a
+    model.
+
+    Args:
+        block : model to be studied
+
+    Returns:
+        A ComponentSet including all GreyBox Block components in block
+        (including block itself)
+    """
+    block_set = ComponentSet()
+    for grey_box in greybox_block_set(block):
+        if grey_box.active:
+            block_set.add(grey_box)
+
+    return block_set
+
+
+def deactivated_greybox_block_set(block):
+    """
+    Function to return ComponentSet of deactivated Greybox Blocks components in a
+    model.
+
+    Args:
+        block : model to be studied
+
+    Returns:
+        A ComponentSet including all GreyBox Block components in block
+        (including block itself)
+    """
+    return greybox_block_set(block) - activated_greybox_block_set(block)
+
+
+def number_deactivated_greybox_block(block):
+    """
+    Function to return a Number of deactivated Greybox Blocks components in a
+    model.
+
+    Args:
+        block : model to be studied
+
+    Returns:
+        number of deactivated greybox blocks
+    """
+    return len(deactivated_greybox_block_set(block))
+
+
+def number_greybox_blocks(block):
+    """
+    Function to return a Number of activated Greybox Blocks components in a
```

Comment:
```suggestion
    Function to return a number of Greybox Blocks components in a
```

Reply:
@avdudchenko since this will show up in the documentation, I figured this docstring should be unique when comparing with the docstring for `number_activated_greybox_blocks`.